mesonbuild/mconf.py

```diff
@@ -36,6 +36,9 @@ def add_arguments(parser: 'argparse.ArgumentParser') -> None:
     parser.add_argument('builddir', nargs='?', default='.')
     parser.add_argument('--clearcache', action='store_true', default=False,
                         help='Clear cached state (e.g. found dependencies)')
+    parser.add_argument('--allow-unknown-options', action='store_false',
+                        dest='error_unknown',
+                        help='Do *not* fail if unknown build options are encountered')
```

Comment:
I think in mconf.py it should always be fatal, you set options after configuration, the exact list of available options is known at this point.